### PR TITLE
Add explanation of logger output

### LINKED ISSUE #662: Add doc to explain logger output
### 📚 Documentation

Explain the output of the logger for each algorithm.

### Checklist

- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)

### LINKED ISSUE #739: [Question] Where to find the definitions of the terms shown in training?
**Important Note: We do not do technical support, nor consulting** and don't answer personal questions per email.
Please post your question on the [RL Discord](https://discord.com/invite/xhfNqQv), [Reddit](https://www.reddit.com/r/reinforcementlearning/) or [Stack Overflow](https://stackoverflow.com/) in that case.

### Question

I was looking for documentation for definitions of the terms used in training, e.g.

```
-----------------------------------------
| time/                   |             |
|    fps                  | 2447        |
|    iterations           | 3           |
|    time_elapsed         | 30          |
|    total_timesteps      | 73728       |
| train/                  |             |
|    approx_kl            | 0.012046688 |
|    clip_fraction        | 0.124       |
|    clip_range           | 0.2         |
|    entropy_loss         | -2.28       |
|    explained_variance   | 0.58        |
|    learning_rate        | 0.0003      |
|    loss                 | 0.31        |
|    n_updates            | 20          |
|    policy_gradient_loss | -0.0174     |
|    value_loss           | 1.39        |
-----------------------------------------
```

I could not find anything. As an example for value_loss all I got was a link to the source code for PPO which says

```
# Value loss using the TD(gae_lambda) target
value_loss = F.mse_loss(rollout_data.returns, values_pred)
```

This gives a clue but doesn't tell me the whole story.  Is there any documentation for these terms I am missing?

### PR DESCRIPTION
**Pull request by araffin**
## Description

closes #739
closes #662

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: docs/common/logger.rst [hunk 1]
```diff
@@ -28,5 +28,53 @@ Available formats are ``["stdout", "csv", "log", "tensorboard", "json"]``.
...
+- ``mean_ep_length``: Mean episode length
+- ``mean_reward``: Mean episodic reward (during evaluation), an ``EvalCallback`` is required to compute that value
+- ``success_rate``: Mean success rate during evaluation (1.0 means 100% success), the environment info dict must contain an ``is_success`` key to compute that value
+
+rollout/
+^^^^^^^^
+- ``ep_len_mean``: Mean episode length (averaged over 100 episodes)
+- ``ep_rew_mean``: Mean episodic training reward (averaged over 100 episodes), a ``Monitor`` wrapper is required to compute that value
+- ``exploration_rate``: Current value of the exploration rate when using DQN, it corresponds to the fraction of actions taken randomly (epsilon of the "epsilon-greedy" exploration)
+- ``success_rate``: Mean success rate during training (averaged over 100 episodes), you must pass an extra argument to the ``Monitor`` wrapper to log that value (``info_keywords=("is_success",)``)
```
Comment (Miffyli) [inline docs/common/logger.rst:49]:
```suggestion
- ``success_rate``: Mean success rate during training (averaged over 100 episodes), you must pass an extra argument to the ``Monitor`` wrapper to log that value (``info_keywords=("is_success",)``) and provide ``info["is_success"]=True/False`` on the final step of the episode
```

File: docs/common/logger.rst [hunk 2]
```diff
@@ -28,5 +28,53 @@ Available formats are ``["stdout", "csv", "log", "tensorboard", "json"]``.
...
+train/
+^^^^^^
+- ``actor_loss``: Current value for the actor loss for off-policy algorithms
+- ``approx_kl``: approximate mean KL divergence between old and new policy (for PPO), it is an estimation of how much changes happened in the update
+- ``clip_fraction``: mean fraction of surrogate loss that was clipped (above ``clip_range`` threshold) for PPO.
+- ``clip_range``: Current value of the clipping factor for the surrogate loss of PPO
+- ``critic_loss``: Current value for the critic function loss for off-policy algorithms, usually error between value function output and TD(0), temporal difference estimate
+- ``ent_coef``: Current value of the entropy coefficient (when using SAC)
+- ``ent_coef_loss``: Current value of the entropy coefficient loss (when using SAC)
+- ``entropy_loss``: Mean value of the entropy loss (inversely proportional to the entropy of the policy)
```
Comment (Miffyli) [inline docs/common/logger.rst:68]:
```suggestion
- ``entropy_loss``: Mean value of the entropy loss (negative of the average policy entropy)
```

**Review by Miffyli — COMMENTED** [review]
Thanks for doing this! Some small comments :9

File: docs/common/logger.rst [hunk 3]
```diff
@@ -28,5 +28,53 @@ Available formats are ``["stdout", "csv", "log", "tensorboard", "json"]``.
...
+eval/
+^^^^^
+- ``mean_ep_length``: Mean episode length
+- ``mean_reward``: Mean episodic reward (during evaluation), an ``EvalCallback`` is required to compute that value
+- ``success_rate``: Mean success rate during evaluation (1.0 means 100% success), the environment info dict must contain an ``is_success`` key to compute that value
+
+rollout/
+^^^^^^^^
+- ``ep_len_mean``: Mean episode length (averaged over 100 episodes)
+- ``ep_rew_mean``: Mean episodic training reward (averaged over 100 episodes), a ``Monitor`` wrapper is required to compute that value
```
[the remark this answers is not in the document]
Comment (duttashi) [reply docs/common/logger.rst:47]: From a statistical perspective, `median` is a more robust measure than `mean` for averaging some numeric values. I think episodic length mean's and episodic reward means could be measured in terms of median. This way, the average value will not be susceptible to high reward/length values.

[the remark this answers is not in the document]
Comment (araffin) [reply docs/common/logger.rst:47]: That's a bit offtopic w.r.t. that PR but to answer you, the training reward is only a proxy of the performance and should not be used for evaluation/ proper comparison (that's what we have the `EvalCallback` that also reports standard deviation). And median is also not a perfect metric as it may not reflect enough the distribution of scores (you can read more about evaluation [here](https://araffin.github.io/post/rliable/) where IQM is suggested as a replacement), for instance you can change half the scores to zero without changing the median.

File: docs/common/logger.rst [hunk 4]
```diff
@@ -28,5 +28,53 @@ Available formats are ``["stdout", "csv", "log", "tensorboard", "json"]``.
...
   model.learn(10000)
 
+
+Explanation of logger output
+----------------------------
+
+You can find below short explanations of the value logged in Stable-Baselines3 (SB3).
+Depending on the algorithm used and of the wrappers/callbacks applied, SB3 only logs a subset of those keys during training.
+
+eval/
```
[the remark this answers is not in the document]
Comment (araffin) [reply docs/common/logger.rst:38]: yes

[the remark this answers is not in the document]
Comment (Miffyli) [reply docs/common/logger.rst:38]: Might want to mention that `EvalCallback` is required for any of the `eval/` stats, in that case ^^ (sorry, forgot to suggest this in the original message).

[the remark this answers is not in the document]
Comment (araffin) [reply docs/common/logger.rst:69]: no worry, I understood the underlying remark and did it ;)

**Review by Miffyli — APPROVED** [review]
LGTM, cheers :)
